### PR TITLE
General clean-up PR

### PR DESCRIPTION
**Pull request by xylar**
This merge performs various types of cleanup to the repo.

It sets the level of output from `ncremap` to a near minimal level.  It suppresses all non-error output from `ESMF_RegridWeightGen`.  

A new function, `shared.io.write_netcdf`, is introduced for writing `xarray` data sets with finite `_FillValue`, taken by default from the `netCDF4` package.  This change makes various NCO calls happier and suppresses warnings about non-finite `_FillValue`.

With these changes, calls to `ncremap` produce output like:
```
Input #00: /media/xylar/extra_data/analysis/output/GMPAS-QU240/simplify_ncremap_output/clim/mpas/mld_oQU240_JAS_years0002-0005.nc
Map/Wgt  : /media/xylar/extra_data/analysis/output/GMPAS-QU240/simplify_ncremap_output/mapping/map_climatologyMapMLD_oQU240_to_0.5x0.5degree_bilinear.nc
Regridded: /media/xylar/extra_data/analysis/output/GMPAS-QU240/simplify_ncremap_output/clim/mpas/regridded/mld_oQU240_to_0.5x0.5degree_JAS_years0002-0005.nc
```

In the config file, the `overwrite*` options have been removed and the behavior is now that climatologies and cached time series are not overwritten.

Variables and config options named "regrid*" have been renamed "remap*" for consistent naming throughout the repo.

"Last Modified" entries in docstring have been removed.  These are hard to maintain consistently and the information about when a file or function was last modified is typically available through the "blame" view on GitHub.

Removes config paths to individual mapping files. Instead, the config file supplies an option for a directory with existing mapping files, requiring that all mapping files follow the MPAS-Analysis naming
conventions.  If no mapping directory is supplied or mapping files are not found in the supplied directory, they are created in the mappingSubdirectory of the output directory.

### CODE REVIEW

**Comment by xylar** [comment]
This simple PR is intended to address #142.

**Comment by xylar** [comment]
I actually figured out a good solution for writing `xarray` files to NetCDF with finite `_FillValue`.  I'd like to include that in this PR but it really only makes sense to do that after #222, since we make various changes to when and how we write out NetCDF data sets in that PR.  So I'm going to rebase onto that PR.

**Comment by xylar** [comment]
@milenaveneziani, I decided to make this PR independent of #222, in part because I have a different approach for that PR that I'm going to use instead.  So please take a look at these clean up options, give it a quick test, and merge if/when you think it's ready.

**Comment by xylar** [comment]
@pwolfram, if you have time to review this, that would be great.  If not, please remove yourself as a reviewer.

File: mpas_analysis/shared/io/write_netcdf.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    ------
+    Xylar Asay-Davis
+
+    '''
+    encodingDict = {}
+    for variableName in ds:
+        dtype = ds[variableName].dtype
+        for fillType in fillValues:
+            if dtype == numpy.dtype(fillType):
+                encodingDict[variableName] = \
```
Comment (pwolfram) [inline mpas_analysis/shared/io/write_netcdf.py]: Nice!  Presumably this means that if the vtk extractor tool is applied to this netcdf file then plotting in ParaView can automatically account for NaN values, correct?

Comment (xylar) [inline mpas_analysis/shared/io/write_netcdf.py]: I don't think Paraview notices _FilllValue correctly.  At least, I haven't had much luck with it.  But I also don't like how it treats the geometry so I use the paraview extractor tool instead.  But you should try it out and see what you find.

**Comment by xylar** [comment]
Thanks @pwolfram. No, I don't think there's anything specific that needs your attention as long as you're good with the list clean up in the description of the PR.

File: mpas_analysis/shared/io/write_netcdf.py (hunk 1 above)
Comment (pwolfram) [inline mpas_analysis/shared/io/write_netcdf.py]: Does this mean that the "paraview extractor tool" you wrote handles _FillValue correctly when plotting in paraview, e.g., no `-1E34` numbers?  Not a big deal, just curious if this was a nice side benefit of the PR.

Comment (xylar) [inline mpas_analysis/shared/io/write_netcdf.py]: @pwolfram, no, the paraview extractor tool doesn't fix the fact that MPAS doesn't provide `_FillValue` and neither does this PR.  This is fixing things the other direction, where xarray likes to use `_FillValue=NaN` and many other applications, notably NCO, don't like that choice.

**Comment by milenaveneziani** [comment]
@xylar: if you rebase this, I will give it a quick test and merge it. thanks.

**Comment by xylar** [comment]
Thanks @milenaveneziani.  I just rebased.

**Comment by milenaveneziani** [comment]
output/messages look very clear now. Thanks @xylar.

**Comment by xylar** [comment]
Great, thanks!

**Pull request merged**